src/main/java/io/stargate/sgv2/jsonapi/config/feature/DataApiFeatures.java

```diff
@@ -0,0 +1,43 @@
+package io.stargate.sgv2.jsonapi.config.feature;
+
+import io.stargate.sgv2.jsonapi.api.request.DataApiRequestInfo;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Accessor for combined state of feature flags; typically based on static configuration (with its
+ * overrides) and possible per-request settings.
+ */
+public class DataApiFeatures {
+  private final Map<DataApiFeatureFlag, Boolean> fromConfig;
+  private final DataApiRequestInfo.HttpHeaderAccess httpHeaders;
+
+  DataApiFeatures(
+      Map<DataApiFeatureFlag, Boolean> fromConfig,
+      DataApiRequestInfo.HttpHeaderAccess httpHeaders) {
+    this.fromConfig = (fromConfig == null) ? Collections.emptyMap() : fromConfig;
```

Comment:
what is the situation where the fromConfig is null ? could this lead to a subtle bug where the config is missed ?

consider enforcing non-null

nit: also, consider making ctor private if we want to construct via the static factories.

Reply:
Can/will change to `private`. We get "config" from Quarkus, but I can add `null` check in caller -- but I think it probably better be there just since config mapping might indeed get `null` if our `application.yaml` has empty value (which is reasonable if we do not define any enabled features).